tests/test_forecast.py

```diff
@@ -0,0 +1,304 @@
+        with pytest.raises(
+            ValueError, match="time index of the forecasts is not the same"
+        ):
+            _check_forecast_data(forecasts)
+
+    def test_valid_forecast_data(self) -> None:
+        """Test that valid forecast data returns the correct values."""
+        index: DatetimeIndex = pd.date_range("2023-01-01", periods=10, freq="D")
+        df1: DataFrame = pd.DataFrame(np.random.rand(10, 3), index=index)
+        df2: DataFrame = pd.DataFrame(np.random.rand(10, 3), index=index)
+        forecasts: dict[str, list[DataFrame]] = {"sm1": [df1, df2]}
+        n, tmin, tmax, result_index = _check_forecast_data(forecasts)
+        assert n == 3
+        assert tmin == index[0]
+        assert tmax == index[-1]
+        assert (result_index == index).all()
+
+
+class TestGetOverallMeanAndVariance:
+    index: DatetimeIndex
+    df: DataFrame
+
+    def setup_method(self) -> None:
+        """Set up test data for get_overall_mean_and_variance."""
+        self.index = pd.date_range("2023-01-01", periods=5, freq="D")
+        columns: MultiIndex = pd.MultiIndex.from_product(
+            [range(2), range(3), ["mean", "var"]],
+            names=["ensemble_member", "param_member", "forecast"],
+        )
+
+        # Create data with known mean and variance patterns
+        # For ensemble_member=0, param_member=0, mean values are 1.0
+        # For ensemble_member=0, param_member=1, mean values are 2.0
+        # For ensemble_member=0, param_member=2, mean values are 3.0
+        # For ensemble_member=1, param_member=0, mean values are 4.0
+        # For ensemble_member=1, param_member=1, mean values are 5.0
+        # For ensemble_member=1, param_member=2, mean values are 6.0
+        # All var values are 0.5
+
+        data: np.ndarray = np.zeros((5, 12))
+        for i in range(0, 12, 2):
+            data[:, i] = i / 2 + 1  # mean values
+            data[:, i + 1] = 0.5  # variance values
+
+        self.df = pd.DataFrame(data=data, index=self.index, columns=columns)
+
+    def test_overall_mean_calculation(self) -> None:
+        """Test that the overall mean is calculated correctly."""
+        mean, _ = get_overall_mean_and_variance(self.df)
+
+        # Expected mean is average of all mean values (1+2+3+4+5+6)/6 = 3.5
+        assert isinstance(mean, Series)
+        assert len(mean) == len(self.index)
+        assert (mean == 3.5).all()
+
+    def test_overall_variance_calculation(self) -> None:
+        """Test that the overall variance is calculated correctly."""
+        _, var = get_overall_mean_and_variance(self.df)
+
+        # Variance of means: ((1-3.5)² + (2-3.5)² + ... + (6-3.5)²)/5 = 3.5 # ddof=1
+        # Mean of variances: 0.5
+        # Expected total variance = 3.5 + 0.5 = 4
+        assert isinstance(var, Series)
+        assert len(var) == len(self.index)
+        assert (var == 4.0).all()
+
+
+@pytest.fixture
+def mock_model() -> MagicMock:
```

Comment:
why don't we use the model fixture that is already present in the testing routine?

Reply:
`ml_noisemodel`